comet_core/app.py

```diff
@@ -451,14 +452,33 @@ def process_unprocessed_events(self):
             for owner, events in events_by_owner.items():
                 owner_reminder_cadence = source_type_config["owner_reminder_cadence"]
 
-                if any([event.new for event in events]) or self.data_store.check_any_issue_needs_reminder(
-                    owner_reminder_cadence, events
-                ):
+                events_to_remind = []
+                if source_type_config["communication_digest_mode"]:
+                    if any([event.new for event in events]) or self.data_store.check_any_issue_needs_reminder(
+                        owner_reminder_cadence, events
+                    ):
+                        events_to_remind = events
+                else:
+                    fingerprints_to_remind = self.data_store.get_any_issues_need_reminder(
+                        owner_reminder_cadence, events
+                    )
+                    if fingerprints_to_remind:
+                        for e in events:
+                            if e.fingerprint in fingerprints_to_remind:
+                                e.reminder = True
+                                events_to_remind.append(e)
+
+                    for e in events:
+                        if e.new and not e.fingerprint in fingerprints_to_remind:
+                            events_to_remind.append(e)
+
+                if events_to_remind:
                     try:
-                        self._route_events(owner, events, source_type)
-                        self.data_store.update_processed_at_timestamp_to_now(events)
+                        self._route_events(owner, events_to_remind, source_type)
                     except CometCouldNotSendException:
-                        LOG.error(f"Could not send alert to {owner}: {events}")
+                        LOG.error(f"Could not send alert to {owner}: {events_to_remind}")
+
+                self.data_store.update_processed_at_timestamp_to_now(events)
```

Comment:
`self.data_store.update_processed_at_timestamp_to_now(events)` was within the try statement so it would not update the event as sent if it crashed with a `CometCouldNotSendException`.

I think it makes sense to still have it within the try-statement (unless there are other arguments for putting it outside)